src/Mod/Path/libarea/Adaptive.cpp

```diff
@@ -146,6 +146,38 @@ inline DoublePoint GetPathDirectionV(const Path &pth, size_t pointIndex)
 	const IntPoint &p2 = pth.at(pointIndex);
 	return DirectionV(p1, p2);
 }
+	
+bool doFilter(Path &ppg) {
+        size_t i;
+        size_t b;
+        size_t start;
+        size_t end;
+        for(i=0;i<ppg.size();i++) {
+            if (i==ppg.size()-1) {
+                start=0;
+                end=ppg.size()-2;
+            } else {
+                start=i+1;
+                end=ppg.size();
+            }
+            for(b=start;b<end;b++) {
+                if (abs(ppg[i].X-ppg[b].X)<2 && abs(ppg[i].Y-ppg[b].Y)<2) {
+                    ppg.erase(ppg.begin()+b);
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
There are a lot of C-style array accesses here, with the attendant overrun dangers. Your intent would be expressed more clearly using std::find_if().

Reply:
If you have a better idea here please propose

                if (abs(ppg[i].X-ppg[b].X)<2 && abs(ppg[i].Y-ppg[b].Y)<2) {
                    ppg.erase(ppg.begin()+b);
                    return true;
                }

actually I think there could be an else / break as well because only direct nearby values in a row should matter if a similar value occurs at another position it should be left untouched, also I think I'd rather use modulo to go over the array.

I have never used std::find_if, I think it's too generic for that isn't it? (it might possibly describe the current filter, but it might have an issue with what I just wrote in the previous block).

Reply:
I suggest something like (untested!):
```
bool doFilter(Path& p) {
	for (auto start = p.begin(); start != p.end(); ++start) {
		auto IsClose = [start](const IntPoint& testPoint) -> bool {
			return (abs(start->X - testPoint.X) < 2 && abs(start->Y - testPoint.Y) < 2);
		};
		auto closePoint = std::find_if(start, p.end(), IsClose);
		if (closePoint != p.end()) {
			p.erase(closePoint); // In C++20 we should replace with erase_if
			return true;
		}
	}
	return false;
}
```

Reply:
1. you will have to loop over to check against the first item as well, that's why I'd rather use modulo for the iteration actually.
2. this would remove any nearby value within the entire array (yes the same way as my original code does, but I think there's no need to)
eg. X 0,1,1,2,1,1,1,3
this is very simplified because the code pairs x and y; but the code will remove all 1 from the list (I think about edges that loop back to a nearby position like eg. 8 where the center is not touching but very close)
only 0,x,x,2,1,x,x,3 should be left; don't remove the 1 after 2

Reply:
Why would you have to loop over? If a is close to b, then b is close to a. No need to double-check every value.

Reply:
the last vertex needs to be checked against the first, second, etc. one as well. values that differ >1 are boundaries and shouldn't be crossed for removing (as with your code suggestion, and as with my patch -- you can consider both as incomplete)

Reply:
The last vertex is checked against the first: when you check the first. This solution is sounding increasingly ad hoc, perhaps it's best to step back and rethink the goals.

Reply:
I will check within the next days, I still have to finish some other projects first.
I'll patch it into my local FreeCAD version and run my object through the code to see if things are okay.

I think I was a bit confused when I wrote that (I'm into a completely different topic here which requires my attention..)